types/mui-datatables/index.d.ts

```diff
@@ -278,91 +521,266 @@ export type MUIDataTableOptions = Partial<{
+    /**
+     * Show/hide the select all/deselect all checkbox header for selectable rows.
+     * @defaultValue true
+     */
     selectableRowsHeader: boolean;
+    /**
+     * Hides the checkboxes that appear when selectableRows is set to "multiple" or "single".
+     * Can provide a more custom UX, especially when paired with selectableRowsOnClick.
+     * @defaultValue false
+     */
     selectableRowsHideCheckboxes: boolean;
+    /**
+     * Enable/disable select toggle when row is clicked.
+     * When False, only checkbox will trigger this action.
+     * @defaultValue false
+     */
     selectableRowsOnClick: boolean;
+    /**
+     * Controls the visiblity of the Select Toolbar.
+     * Options:
+     * - 'replace': Select toolbar replaces default toolbar.
+     * - 'above': Appears above the defualt toolbar.
+     * - 'none': Select Toolbar never appears
+     *
+     * @defaultValue replace
+     */
     selectToolbarPlacement: 'replace' | 'above' | 'none';
+    /**
+     * Enable remote data source
+     * @defaultValue fale
+     */
     serverSide: boolean;
+    /**
+     * Is called for each filter chip and allows you to place custom props on a filter chip.
+     * @link https://github.com/gregnb/mui-datatables/blob/master/examples/customize-filter/index.js
+     */
     setFilterChipProps: (colIndex: number, colName: string, data: ReadonlyArray<any[]>) => MUIDataTableChip;
+    /**
+     * Is called for each row and allows you to return custom props for this row based on its data.
+     * @link https://github.com/gregnb/mui-datatables/blob/master/examples/customize-styling/index.js
+     */
     setRowProps: (row: any[], dataIndex: number, rowIndex: number) => object;
+    /**
+     * Is called for the table and allows you to return custom props for the table based on its data.
+     * @link https://github.com/gregnb/mui-datatables/blob/master/examples/customize-styling/index.js
+     */
     setTableProps: () => object;
+    /**
+     * Enable/disable sort on all columns.
+     * @defaultValue true
+     */
     sort: boolean;
+    /**
+     * Enable/disable alphanumeric sorting of filter lists.
+     * @defaultValue true
+     */
     sortFilterList: boolean;
+    /**
+     * Sets the column to sort by and its sort direction.
+     * To remove/reset sorting, input in an empty object.
+     * The object options are the column name and the direction.
+     * @shape { name: string, direction: 'asc' | 'desc' }
+     * @link https://github.com/gregnb/mui-datatables/blob/master/examples/customize-columns/index.js
+     */
     sortOrder: MUISortOptions;
+    /**
+     * A string that is used internally for identifying the table.
+     * It's auto-generated, however, if you need it set to a custom value (ex: server-side rendering), you can set it via this property.
+     * @defaultValue auto-generated string
+     */
+    tableId: string;
+    /**
+     * CSS string for the height of the table (ex: '500px', '100%', 'auto').
```

Comment:
```suggestion
     * CSS string for the height of the table.
     * @example '500px'
     * @example '100%'
     * @example 'auto'
```

Lets use proper example tags